DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js

```diff
@@ -0,0 +1,77 @@
+//
+//  checkoutScanner.js
+//
+//  Created by Rebecca Stankus on 10/10/17.
+//  Copyright 2017 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+//  This script acts on the scanner zone to pull up an item's marketplace page to enable purchasing.
+(function() {
+    
+    var TABLET = Tablet.getTablet("com.highfidelity.interface.tablet.system");
+    var MARKET_PLACE_ITEM_URL_PREFIX = 'https://metaverse.highfidelity.com/marketplace';
+    var MARKETPLACES_INJECT_SCRIPT_URL = ScriptDiscoveryService.defaultScriptsPath + "/system/html/js/marketplacesInject.js";
+    var SCAN_RADIUS = 0.15; // meters
+    var OVERLAY_PREFIX = 'MP';
+    
+    var interval;
+    var properties;
+    var checkoutZoneID;
+
+    var scannedMPOverlays = {};
+
+    var onEntityAdded = function(entityID) {
+        // print('entity added: ' + entityID);
+        var newItemProperties = Entities.getEntityProperties(entityID, ['marketplaceID', 'clientOnly', 'owningAvatarID', 'lastEditedBy']);
+        // print(MyAvatar.sessionUUID);
+        //  print(newItemProperties.lastEditedBy);
+        if (/* newItemProperties.clientOnly && newItemProperties.owningAvatarID === MyAvatar.sessionUUID && */
+            (newItemProperties.lastEditedBy === MyAvatar.sessionUUID || newItemProperties.lastEditedBy === undefined) &&
+            newItemProperties.marketplaceID !== "") {
+            
+            // print('Entity of interest added.');
+            if (scannedMPOverlays[newItemProperties.marketplaceID] !== undefined) {
+                // print('scannedMPOverlays contains entity of interest.');
+                var overlayID = scannedMPOverlays[newItemProperties.marketplaceID];
+                Entities.callEntityMethod(checkoutZoneID, 'replicaCheckedOut', [overlayID, entityID]);
+                // print('called the entity method.');
+                delete scannedMPOverlays[newItemProperties.marketplaceID];
+            }
+        }
+    };
+    
+    this.preload = function(entityID) {
```

Comment:
Was there an explicit decision made to not have a single script spawn these together with one the parent of the other? I find the user data approach introduces more risk compared to parentID property.

Reply:
I don't recall parenting zones to zones before, if it works then that would be good, lets see